### DISCUSSION TITLE
Fully silent update for standard user

### DISCUSSION THREAD

**Discussion by orranbass** [sparkle-project/Sparkle#2265] [Q&A]
I have a sandbox app and use Sparkle 2.3 version. I know that sparkle can be fully silent when my app running with root permissions but I can't do it.
I thought to start Sparkle from my System Extension that my app included and it's running with root permissions but I'm getting this error when I calling updater.start():
`SUUpdater -retriveFeedURL:error: must be called on the main thread for ...`
I can't show the user any permission alert so privileged helper tool can't help me either.
How can I run Sparkle with standard user permission and don’t show the user any alerts?
Thanks

**Comment by zorgiepoo** [comment]
That error is unrelated to running as root. Sparkle APIs (just like UI APIs) must be called on the main thread, not a background thread.

Running apps or UI code or AppKit code as root is not properly supported by the OS. Sparkle "may" work but it's not really supported in that context. (That said, Sparkle can support running as root if the calling process has no AppKit/UI usage and a custom non-ui `SPUUserDriver` is used, similar to sparkle-cli). #2169 has some other details.

**Reply by orranbass** [reply]
I included Sparkle framework in my system extension, can I run Sparkle-cli from the framework or I need to install also the Sparkle app in the user machine?

**Reply by zorgiepoo** [reply]
I'm not that familiar with system extensions but I assume they run as root and have a normal runloop lifecycle. You have two potential choices. Embed sparkle-cli inside your extension and spawn it, or embed Sparkle.framework and set up a `SPUUpdater` with your own non-UI driven `SPUUserDriver`. This is what [sparkle-cli ends up doing](https://github.com/sparkle-project/Sparkle/tree/2.x/sparkle-cli) itself.
